Add Raft metrics to Raft config doc

joealewine opened the pull request:
Adding two metrics to Raft configuration doc to encourage
users to monitor their active nodes as compared to the
total number of nodes in the cluster

Change-Id: Ica35ee1a0a2feea6511cfd56fcee1bdf429b1334
Signed-off-by: joe-alewine <Joe.Alewine@ibm.com>

#### Type of change

- Documentation update

#### Description

Adding two metrics to Raft configuration doc to encourage
users to monitor their active nodes as compared to the
total number of nodes in the cluster

Commit:
Add Raft metrics to Raft config doc

Adding two metrics to Raft configuration doc to encourage
users to monitor their active nodes as compared to the
total number of nodes in the cluster

Change-Id: Ica35ee1a0a2feea6511cfd56fcee1bdf429b1334
Signed-off-by: joe-alewine <Joe.Alewine@ibm.com>

joealewine commented:
@wlahti 

Took a shot at explaining this. Let me know if I got close enough to the mark.

wlahti commented on docs/source/raft_configuration.md:
```diff
@@ -317,6 +317,12 @@ monitor:
    they can be shared with the consenter set. If this value begins to climb, this
    node may not be able to participate in consensus (which could lead to a
    service interruption for this node and possibly the network).
+* `consensus_etcdraft_cluster_size` and `consensus_etcdraft_active_nodes`: these
```
We should also mention that these metrics are at the channel level.

wlahti commented on docs/source/raft_configuration.md:
```diff
@@ -317,6 +317,12 @@ monitor:
    node may not be able to participate in consensus (which could lead to a
    service interruption for this node and possibly the network).
+* `consensus_etcdraft_cluster_size` and `consensus_etcdraft_active_nodes`: these
+   metrics help track the "active" nodes (which, as it sounds, are the nodes that
```
Missing the closing ).

wlahti reviewed (commented):
Thanks, Joe! Looks pretty good overall, just needs some updating to mention that we're talking about the channel level here.

joealewine commented:
> Thanks, Joe! Looks pretty good overall, just needs some updating to mention that we're talking about the channel level here.

Thanks for the quick review, Will. I've addressed those comments and pushed an update.

wlahti commented on docs/source/raft_configuration.md:
```diff
@@ -317,6 +317,12 @@ monitor:
+* `consensus_etcdraft_cluster_size` and `consensus_etcdraft_active_nodes`: these
+   channel metrics help track the "active" nodes (which, as it sounds, are the nodes that
+   are currently contributing to the cluster, as compared to the total number of
+   nodes in the cluster). If the number of active nodes falls below a majority of
```
Not sure if it would be redundant or useful to clarify that "the ordering service will stop processing blocks for the channel". Just quickly reading this makes it sound like the orderer will stop processing _all_ blocks but I'll defer to you based on what would be consistent with our other docs for the orderer/raft/consensus.

joealewine replied:
My assumption was that people would understand that "stop processing blocks" referred to just the channel in question (since it's a channel metric), though unless channels have different sets of consenters it likely would be true for at least several channels. But I'm happy to add a slight clarification here to say "stop processing blocks on the channel".

wlahti reviewed (approved):
Looks good to me now. Thanks, Joe!

pamandrejko reviewed (approved):
Thanks for doing this Joe.

pamandrejko commented:
@Mergifyio backport release-2.2

pamandrejko commented:
@Mergifyio backport release-2.1
